qiskit/pulse/transforms.py

```diff
@@ -236,3 +261,127 @@ def compress_pulses(schedules: List[Schedule]) -> List[Schedule]:
         new_schedules.append(new_schedule)
 
     return new_schedules
+
+
+def _push_append(this: List[interfaces.ScheduleComponent],
+                 other: List[interfaces.ScheduleComponent],
+                 ) -> Schedule:
+    r"""Return a ``this`` with ``other`` inserted at the maximum time over
+    all channels shared between ```this`` and ``other``.
+
+   $t = \textrm{max}({x.stop\_time |x \in self.channels \cap schedule.channels})$
+
+    Args:
+        this: Input schedule to which ``other`` will be inserted.
+        other: Other schedule to insert.
+
+    Returns:
+        Joined schedule
+    """
+    this_channels = set(this.channels)
+    other_channels = set(other.channels)
+    shared_channels = list(this_channels & other_channels)
+    ch_slacks = [this.stop_time - this.ch_stop_time(channel) + other.ch_start_time(channel)
+                 for channel in shared_channels]
```

Comment:
generator expr possible?

Reply:
I also am having trouble following this. I think I may have to stop reviewing for today.
this: |--------------| dur = 10
d0:   |---------|         dur = 7
this-d0:           |----| dur = 3
other: 
d0:              |-------| start = 3
ch_slacks = 6
Why the word slacks?
After going through the example, I understand it. I will have to review another day to see if there's a way to write this that can make this easier for others to follow, or if I'm just fatigued.

Reply:
A generator won't work in this case as we require the explicit index for the channel lookup.

Reply:
Slack is in reference to timing slack. This is one of the trickiest pieces of code.